Firestore/Swift/Source/Result/Firestore+Result.swift

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import FirebaseFirestore
+
+#if swift(>=5.0)
+
+  extension Firestore {
+    /// Executes the given updateBlock and then attempts to commit the changes applied within an atomic transaction.
+    ///
+    /// The maximum number of writes allowed in a single transaction is 500, but note that each usage of `FieldValue.serverTimestamp()`, `FieldValue.arrayUnion()`, `FieldValue.arrayRemove()`, or `FieldValue.increment()` inside a transaction counts as an additional write.
+    ///
+    /// In the updateBlock, a set of reads and writes can be performed atomically using the `Transaction` object passed to the block. After the updateBlock is run, Firestore will attempt to apply the changes to the server. If any of the data read has been modified outside of this transaction since being read, then the transaction will be retried by executing the updateBlock again. If the transaction still fails after 5 retries, then the transaction will fail.
+    ///
+    /// Since the updateBlock may be executed multiple times, it should avoiding doing anything that would cause side effects.
+    ///
+    /// Any value maybe be returned from the updateBlock. If the transaction is successfully committed, then the completion block will be passed that value. The updateBlock also has an `Error` out parameter. If this is set, then the transaction will not attempt to commit, and the given error will be passed to the completion block.
```

Comment:
Change

> The updateBlock also has an `Error` out parameter. If this is set, then the transaction will not attempt to commit, and the given error will be passed to the completion block.

to

> If the updateBlock throws an error, then the transaction will not attempt to commit, and the thrown error will be passed to the completion block in `result`.

... or something similarly phrased.

Reply:
Done.